src/Nonlinear/ReverseAD/mathoptinterface_api.jl

```diff
@@ -216,13 +216,16 @@ function MOI.eval_constraint_jacobian(d::NLPEvaluator, J, x)
 end
 
 function MOI.eval_constraint_jacobian_product(d::NLPEvaluator, y, x, w)
+    _reverse_mode(d, x)
```

Comment:
I am not sure to understand why we need `_reverse_mode` here. I would say using forward mode would be more appropriate to evaluate Jacobian-product.

Reply:
`_reverse_mode` is a just a caller for the process which updates the gradients of each expression.

> I would say using forward mode would be more appropriate to evaluate Jacobian-product.

That's potentially true, but that's a much bigger change. For now, this is a clear win because it removes some allocations.

Reply:
I see. I agree with you: if this is not direct to implement, we should merge this PR first. How about opening a separate issue to track this, or add a `TODO` comment in the code?

Reply:
I'll mark it as improves #1903, but not closes.